content/algorithms/dag/index.md

```diff
@@ -0,0 +1,216 @@
+print(nx.is_directed_acyclic_graph(clothing_graph))
+```
+
+But, the `triangle_graph` is not a DAG.
+
+```{code-cell} ipython3
+# check if the `triangle_graph` is DAG
+print(nx.is_directed_acyclic_graph(triangle_graph))
+```
+
+`triangle_graph` has a cycle:
+
+```{code-cell} ipython3
+# find cycle in the `triangle_graph`
+print(nx.find_cycle(triangle_graph))
+```
+
+### Applications
+
+Directed acyclic graphs representations of partial orderings have many applications in scheduling
+for systems of tasks with ordering constraints.
+An important class of problems of this type concern collections of objects that need to be updated,
+such as the cells of a spreadsheet after one of the cells has been changed,
+or the object files of a piece of computer software after its source code has been changed.
+In this context, a dependency graph is a graph that has a vertex for each object to be updated,
+and an edge connecting two objects whenever one of them needs to be updated earlier than the other.
+A cycle in this graph is called a circular dependency, and is generally not allowed,
+because there would be no way to consistently schedule the tasks involved in the cycle.
+Dependency graphs without circular dependencies form DAGs.
+
+A directed acyclic graph may be used to represent a network of processing elements.
+In this representation, data enters a processing element through its incoming edges
+and leaves the element through its outgoing edges.
+For instance, in electronic circuit design, static combinational logic blocks
+can be represented as an acyclic system of logic gates that computes a function of an input,
+where the input and output of the function are represented as individual bits.
+
+### Definition
+
+Directed acyclic graph ("DAG" or "dag") is a directed graph with no directed cycles.
+That is, it consists of vertices and edges (also called arcs), with each edge directed from one vertex to another,
+such that following those directions will never form a closed loop.
+
+A directed graph is a DAG if and only if it can be topologically ordered
+by arranging the vertices as a linear ordering that is consistent with all edge directions.
+
+## Topological sort
+
+### Example
+
+```{code-cell} ipython3
+# get a topological sort of a dag
+print(list(nx.topological_sort(clothing_graph)))
+```
+
+### Applications
+
+The canonical application of topological sorting is in scheduling a sequence of jobs
+or tasks based on their dependencies.
+The jobs are represented by vertices, and there is an edge from $u$ to $v$
+if job $u$ must be completed before job $v$ can be started
+(for example, when washing clothes, the washing machine must finish before we put the clothes in the dryer).
+Then, a topological sort gives an order in which to perform the jobs.
+
+A closely related application of topological sorting algorithms
+was first studied in the early 1960s in the context of the PERT technique for scheduling in project management.
```

Comment:
It's always good to fully introduce an acronym the first time you use it. In this case, a link to the PERT wiki would be sufficient I think:

```suggestion
was first studied in the early 1960s in the context of the [PERT technique](https://en.wikipedia.org/wiki/Program_evaluation_and_review_technique) for scheduling in project management.
```